unified-prototype/unified-plugin/plugin-android/src/main/java/org/gradle/api/experimental/android/extensions/ComposeSupport.java

```diff
@@ -0,0 +1,90 @@
+package org.gradle.api.experimental.android.extensions;
+
+import com.android.build.api.dsl.CommonExtension;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.dsl.DependencyHandler;
+import org.gradle.api.experimental.android.AndroidSoftware;
+import org.gradle.api.provider.Provider;
+import org.jetbrains.kotlin.gradle.dsl.KotlinJvmOptions;
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.*;
+
+public final class ComposeSupport {
+    private ComposeSupport() { /* Not instantiable */ }
+
+    @SuppressWarnings("UnstableApiUsage")
+    public static void configureCompose(Project project, AndroidSoftware dslModel, CommonExtension<?, ?, ?, ?, ?, ?> androidLib) {
+        if (dslModel.getCompose().getEnabled().get()) {
+            androidLib.getBuildFeatures().setCompose(true);
+
+            androidLib.getComposeOptions().setKotlinCompilerExtensionVersion("1.5.12");
+
+            DependencyHandler dependencies = project.getDependencies();
+            dependencies.add("implementation", dependencies.platform("androidx.compose:compose-bom:2024.02.02"));
```

Comment:
Can we use the new API for this? e.g.

`dslModel.getDependencies().getImplementation().add(...)`